lib/burnside/intrinsics.cc

```diff
@@ -72,8 +72,17 @@ class MathRuntimeLibrary {
   Map library;
 };
 
-/// enum used to templatized code gen for intrinsics that are alike.
+/// enum used to templatize code gen for intrinsics that are alike.
 enum class Extremum { Min, Max };
+// There are many ways to deal with NaNs in Min and Max.
+// Existing behaviors are listed below.
+enum class ExtremumBehavior {
+  IeeeMinMaxNum /* IEEE 754-2008 */,  // TODO: Not implemented
+  IeeeMinMaximumNumber /* IEEE 754-2018 */,  // gfortran
+  IeeeMinMaximum /* IEEE 754-2018 */,
```

Comment:
Hmm. These three enums are very similar. I wonder if that will be confusing.

Reply:
They are very similar, but they are taken from the actual IEEE names:
- `IeeeMinMaxNum `: `minNum` / `maxNum` (defined in section 53.1 of IEEE 754-2008).
- `IeeeMinMaximumNumber `: `minimumNumber`/`maximumNumber` (defined in section 9.6 of IEEE 754-2019 (I needed to fix the year in the comment, I used the draft year: 2018))
- `IeeeMinMaximum `: `minimum`/`maximum` (also defined in section 9.6 of IEEE 754-2019)

Do you see less confusing names for the enums ?

Reply:
lol. In this case, the standard just lets one fall on his face.

I don't see any great way out. But thanks for the explanation as to the source of these names. :)